src/brs/http/APIServlet.java

```diff
@@ -236,13 +236,23 @@ void addErrorMessage(HttpServletResponse resp, JsonElement msg) throws IOExcepti
   private static Map<String, PrimitiveRequestHandler> primitiveRequestHandlers;
 
   @Override
-  protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws IOException {
-    process(req, resp);
+  protected void doGet(HttpServletRequest req, HttpServletResponse resp) {
+    try {
+      process(req, resp);
+    } catch (Throwable t) { // We don't want to send exception information to client...
+      resp.setStatus(500);
```

Comment:
Is it safe al ways throw a 500 here? thinking about 4xx errors too.

Reply:
The implementation of the process method *should* handle any expected errors - if an exception was not caught by it we should probably throw a 500.